### PR TITLE
Fix bootup git warnings on arch

### LINKED ISSUE #17604: Load packages in metasploit arch linux
When i run msfconsole in my Arch distribution, previously installed with sudo pacman -S metasploit. I have this line at the moment to run the msfconsole
 error: unable to normalize alternate object path: /srcdest/metasploit-framework/objects

##  Module/Datastore

The following global/module datastore, and database setup was configured before the issue occurred:
<details>
<summary>Collapse</summary>

```
[framework/core]
loglevel=3
```

</details>

##  Database Configuration

The database contains the following information:
<details>
<summary>Collapse</summary>

```
Session Type: postgresql selected, no connection
```

</details>

##  History

The following commands were ran during the session and before this issue occurred:
<details>
<summary>Collapse</summary>

```
11     set loglevel 3
12     debug
```

</details>

##  Framework Errors

The following framework errors occurred before the issue occurred:
<details>
<summary>Collapse</summary>

```
[02/06/2023 12:43:33] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[02/06/2023 12:43:35] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[02/06/2023 12:44:28] [e(0)] core: Failed to connect to the database: No database YAML file
[02/06/2023 12:44:30] [e(0)] core: Dependency for windows/x64/encrypted_shell_reverse_tcp is not supported
[02/06/2023 12:44:30] [e(0)] core: Dependency for windows/encrypted_shell_reverse_tcp is not supported
[02/06/2023 12:44:32] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[02/06/2023 12:44:34] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[02/06/2023 12:50:57] [e(0)] core: Failed to connect to the database: No database YAML file
[02/06/2023 13:09:10] [e(0)] core: Failed to connect to the database: No database YAML file
[02/06/2023 13:13:14] [e(0)] core: Failed to connect to the database: No database YAML file
```

</details>

##  Web Service Errors

The following web service errors occurred before the issue occurred:
<details>
<summary>Collapse</summary>

```
msf-ws.log does not exist.
```

</details>

##  Framework Logs

The following framework logs were recorded before the issue occurred:
<details>
<summary>Collapse</summary>

```
[02/06/2023 12:43:00] [e(0)] core: Failed to connect to the database: No database YAML file
[02/06/2023 12:43:00] [d(0)] core: Created user based module store
[02/06/2023 12:43:03] [e(0)] core: Dependency for windows/x64/encrypted_shell_reverse_tcp is not supported
[02/06/2023 12:43:03] [e(0)] core: Dependency for windows/encrypted_shell_reverse_tcp is not supported
[02/06/2023 12:43:04] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[02/06/2023 12:43:06] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[02/06/2023 12:43:07] [d(0)] core: HistoryManager.push_context name: :msfconsole
[02/06/2023 12:43:20] [d(0)] core: HistoryManager.pop_context name: :msfconsole
[02/06/2023 12:43:29] [e(0)] core: Failed to connect to the database: No database YAML file
[02/06/2023 12:43:31] [e(0)] core: Dependency for windows/x64/encrypted_shell_reverse_tcp is not supported
[02/06/2023 12:43:31] [e(0)] core: Dependency for windows/encrypted_shell_reverse_tcp is not supported
[02/06/2023 12:43:33] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[02/06/2023 12:43:35] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[02/06/2023 12:43:36] [d(0)] core: HistoryManager.push_context name: :msfconsole
[02/06/2023 12:44:17] [d(0)] core: HistoryManager.pop_context name: :msfconsole
[02/06/2023 12:44:28] [e(0)] core: Failed to connect to the database: No database YAML file
[02/06/2023 12:44:30] [e(0)] core: Dependency for windows/x64/encrypted_shell_reverse_tcp is not supported
[02/06/2023 12:44:30] [e(0)] core: Dependency for windows/encrypted_shell_reverse_tcp is not supported
[02/06/2023 12:44:32] [e(0)] core: Dependency for windows/x64/encrypted_reverse_tcp is not supported
[02/06/2023 12:44:34] [e(0)] core: Dependency for windows/encrypted_reverse_tcp is not supported
[02/06/2023 12:44:35] [d(0)] core: HistoryManager.push_context name: :msfconsole
[02/06/2023 12:45:41] [d(0)] core: HistoryManager.pop_context name: :msfconsole
[02/06/2023 12:50:57] [e(0)] core: Failed to connect to the database: No database YAML file
[02/06/2023 12:51:06] [d(0)] core: HistoryManager.push_context name: :msfconsole
[02/06/2023 12:52:00] [d(0)] core: HistoryManager.pop_context name: :msfconsole
[02/06/2023 13:09:10] [e(0)] core: Failed to connect to the database: No database YAML file
[02/06/2023 13:09:17] [d(0)] core: HistoryManager.push_context name: :msfconsole
[02/06/2023 13:13:08] [d(0)] core: HistoryManager.pop_context name: :msfconsole
[02/06/2023 13:13:14] [e(0)] core: Failed to connect to the database: No database YAML file
[02/06/2023 13:13:21] [d(0)] core: HistoryManager.push_context name: :msfconsole
```

</details>

##  Web Service Logs

The following web service logs were recorded before the issue occurred:
<details>
<summary>Collapse</summary>

```
msf-ws.log does not exist.
```

</details>

##  Version/Install

The versions and install method of your Metasploit setup:
<details>
<summary>Collapse</summary>

```
Framework: 6.2.36-dev-bca8374
Ruby: ruby 3.0.5p211 (2022-11-24 revision ba5cf0f7c5) [x86_64-linux]
OpenSSL: OpenSSL 3.0.7 1 Nov 2022
Install Root: /opt/metasploit
Session Type: postgresql selected, no connection
Install Method: Git Clone
```

</details>
6. Replace these instructions and the paragraph above with the output from step 5.

### PR DESCRIPTION
**Pull request by adfoster-r7**
closes https://github.com/rapid7/metasploit-framework/issues/17604

## Verification

Run arch:
```
docker run -it archlinux /bin/bash
```

Install metasploit / vim:
```
echo y | pacman -Sy metasploit
echo y | pacman -Sy vim
```

### Before

Bootup warnings:

```
[root@61c1980f91fb /]# msfconsole -qx 'version; exit'
error: unable to normalize alternate object path: /srcdest/metasploit-framework/objects
Framework: 6.2.36-dev-bca8374
Console  : 6.2.36-dev-bca8374
```

### After 

After manually patching the PR changes with `vim /opt/metasploit/lib/metasploit/framework/version.rb`; No warnings:

```
[root@3cf8733a3890 /]# msfconsole -qx 'version; exit'
Framework: 6.2.36-dev-bca8374
Console  : 6.2.36-dev-bca8374
```

### CODE REVIEW

**Comment by sempervictus** [comment]
Thanks!
There's an active effort by the maintainers right now to re-do how gems (including those in std) are packaged for Arch which is holding up the interpreter version @ `3.0.x` - spoke w/ relevant party in IRC recently, they're hoping to have 3.1 by March with 3.2 to follow soon after.

**Comment by gwillcox-r7** [comment]
Before:

```
[root@bb3ce153ce1a /]# msfconsole -qx 'version; exit'
error: unable to normalize alternate object path: /srcdest/metasploit-framework/objects
Framework: 6.2.36-dev-bca8374
Console  : 6.2.36-dev-bca8374
[root@bb3ce153ce1a /]# 
```

**Comment by gwillcox-r7** [comment]
After:

```
[root@bb3ce153ce1a /]# msfconsole -qx 'version; exit'
Framework: 6.2.36-dev-bca8374
Console  : 6.2.36-dev-bca8374
[root@bb3ce153ce1a /]# 
```

**Review by gwillcox-r7 — APPROVED** [review]

**Comment by gwillcox-r7** [comment]
# Release Notes
A bug when running `msfconsole` and attempting to get the Metasploit version installed has been fixed by improving the way the GitHub version detection logic works.

**Pull request merged**
